### DISCUSSION TITLE
Scaling report generation

### DISCUSSION THREAD

**Discussion by damsleth** [Puzzlepart/did#944] [General]
## TLDR - how do we make reports scalable?

There are multiple aspects to this

* Time to retrieve data from mongoDB
  * Currently around 45 seconds for "Last Year" report (excluding transfer to client ~10s)
* Can we put the JSON-format report retrieved from mongoDB in a table? E.g "reports" (GridFS or blob storage), with a timestamp and name of the report, to make it more quickly retrievable?
* Size sent to client in JSON format after retrieval from mongoDB
  * Currently around 92mb
* Pre-generated XLSX reports server side, to prevent the need for the client to download the whole table
  * xlsx for "Last Year" report is approx 25mb, 1/4 of the JSON retrieved)

**Comment by olemp** [comment]
Oh, I thought we had a discussion for this. Where did it go? Oh, well. Now we have one.

Yes, it doesn't scale well, and it's been well known from the start. No matter what we cache and don't the report will still be huge for loading to the browser. Some backend services for sending the report by email or something?

cc @okms

**Comment by okms** [comment]
![image](https://user-images.githubusercontent.com/170147/115759154-3a1ce880-a3a0-11eb-96ed-0cacf06a0d3c.png)
![image](https://user-images.githubusercontent.com/170147/115759190-43a65080-a3a0-11eb-8b80-12a00e60f0ba.png)
![image](https://user-images.githubusercontent.com/170147/115759242-53be3000-a3a0-11eb-98a7-4a978ae62d90.png)

**Comment by olemp** [comment]
We'll "scale it" by implementing this for now: https://github.com/Puzzlepart/did/pull/1095
